addons/account/models/account_payment_term.py

```diff
@@ -206,17 +201,23 @@ def _compute_terms(self, date_ref, currency, company, tax_amount, tax_amount_cur
             if line.value == 'fixed':
                 term_vals['company_amount'] = sign * company_currency.round(line.value_amount / rate) if rate else 0.0
                 term_vals['foreign_amount'] = sign * currency.round(line.value_amount)
-            elif line == last_percent_line:
-                term_vals['company_amount'] = residual_amount
-                term_vals['foreign_amount'] = residual_amount_currency
             else:
                 line_amount = company_currency.round(total_amount * (line.value_amount / 100.0))
                 line_amount_currency = currency.round(total_amount_currency * (line.value_amount / 100.0))
                 term_vals['company_amount'] = line_amount
                 term_vals['foreign_amount'] = line_amount_currency
-                residual_amount -= line_amount
-                residual_amount_currency -= line_amount_currency
-            pay_term["line_ids"].append(term_vals)
+
+            residual_amount -= term_vals['company_amount']
+            residual_amount_currency -= term_vals['foreign_amount']
+            pay_term['line_ids'].append(term_vals)
+
+        # Add the last line again as a balance line, ignoring its value
+        term_vals = {
+            'date': last_line._get_due_date(date_ref),
+            'company_amount': residual_amount,
+            'foreign_amount': residual_amount_currency,
+        }
+        pay_term['line_ids'].append(term_vals)
```

Comment:
I think you can simplify all of that
```py
        rate = abs(total_amount_currency / total_amount) if total_amount else 0.0
        residual_amount = total_amount
        residual_amount_currency = total_amount_currency
        for i, line in enumerate(self.line_ids):
            term_vals = {
                'date': line._get_due_date(date_ref),
                'company_amount': 0,
                'foreign_amount': 0,
            }
            if i == len(self.line_ids) - 1:
                term_vals['company_amount'] = residual_amount
                term_vals['foreign_amount'] = residual_amount_currency
            elif line.value == 'fixed':
                term_vals['company_amount'] = sign * company_currency.round(line.value_amount / rate) if rate else 0.0
                term_vals['foreign_amount'] = sign * currency.round(line.value_amount)
            else:
                line_amount = company_currency.round(total_amount * (line.value_amount / 100.0))
                line_amount_currency = currency.round(total_amount_currency * (line.value_amount / 100.0))
                term_vals['company_amount'] = line_amount
                term_vals['foreign_amount'] = line_amount_currency
            residual_amount -= term_vals['company_amount']
            residual_amount_currency -= term_vals['foreign_amount']
            pay_term["line_ids"].append(term_vals)

        return pay_term
```

Reply:
Absolutely much nicer 👌 I changed the code as per your suggestion. Thanks!